android/src/ch/opengis/qfield/QFieldOpenExternallyActivity.java

```diff
@@ -0,0 +1,75 @@
+package ch.opengis.qfield;
+
+import java.io.File;
+
+import android.os.Bundle;
+import android.os.Build;
+import android.os.Environment;
+import android.net.Uri;
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+import android.support.v4.content.FileProvider;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.util.Date;
+import java.text.SimpleDateFormat;
+
+public class QFieldOpenExternallyActivity extends Activity{
+    private static final String TAG = "QField Open (file) Externally Activity";
+    private String filePath;
+    private String mimeType;
+    private String tempFileName;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState){
+        Log.d(TAG, "onCreate()");
+        super.onCreate(savedInstanceState);
+
+        filePath = getIntent().getExtras().getString("filepath");
+        Log.d(TAG, "Received filepath: " + filePath);
+        mimeType = getIntent().getExtras().getString("filetype");
+        Log.d(TAG, "Received mimeType: " + mimeType);
+
+        File file = new File(filePath);
+        File cacheFile = new File(getCacheDir(), file.getName());
+
+        //copy file to a temporary file
+        try{
+            copyFile( file, cacheFile );
+        }catch(IOException e){
+            Log.d(TAG, e.getMessage());
```

Comment:
Probably at this point, if you didn't copy the file, you should stop the activity and inform the user that something was wrong

Reply:
okay

The goal is to open an image. This failed before on images we did not had access to (e.g. on demo projects). And it's the preparation for image editing and for open other files like pdf and docx...

Reply:
ok, but ATM since you do a copy of the file you're opening the copy and if you edit it you'll not see the changes in QField. Is that ok?

Reply:
For the edit part I need to receive a result I think, like the camera does.
The edit part did not work, when I just edited a file, since I don't have the right to edit. So the edit part would be kind of this:
- copy the file to cache - like here
- edit the file and pass it to result
- get result

Reply:
ok. so, except for the exception management I already mentioned, everything looks good to me.

Reply:
Perfect. Thanks. Btw. don't you think it would make sense to be able to open the image in read only mode as well?